### PR TITLE
docs: add Java docs and change code style to Google Code Style

### PR DESCRIPTION
**Pull request by arvgord**
(#13)
- Changed code style
- Added Java docs in changed classes

### CODE REVIEW

File: src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java [hunk 1]
```diff
@@ -19,46 +19,60 @@
 import java.util.Collections;
 import java.util.List;
 
+/**
+ * The enum Modality.
+ */
 public enum Modality {
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java:25]: "Utility class for mapping Modality Description to Modality Value" - in general the Javadoc needs to describe what the class does...

File: src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java [hunk 2]
```diff
@@ -19,46 +19,60 @@
...
-    }
+  Modality(String modalityValue) {
+    this.modalityValue = modalityValue;
+  }
 
-    public static List<String> getValues(){
-        return Collections.unmodifiableList(VALUES);
-    }
+  /**
+   * Gets all values.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java]: Returns all Modality Values.

File: src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java [hunk 3]
```diff
@@ -19,46 +19,60 @@
...
+   */
+  public static List<String> getValues() {
+    return Collections.unmodifiableList(VALUES);
+  }
 
-    public String getValue(){
-        return modalityValue;
-    }
+  /**
+   * Gets value.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/Modality.java]: Returns the Modality Value of the Modality Description.

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/DatasetsPathBuilder.java [hunk 4]
```diff
@@ -15,21 +15,35 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.DATASETS;
+import static com.quantumsoft.qupathcloud.dao.Constants.LOCATIONS;
+import static com.quantumsoft.qupathcloud.dao.Constants.PROJECTS;
+import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * DatasetsPathBuilder to build a path for access to Datasets in a chosen Project and Location.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/DatasetsPathBuilder.java]: "DatasetsPathBuilder builds a path to access Datasets in the specified Project and Location." ?

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/DicomStoresPathBuilder.java [hunk 5]
```diff
@@ -15,26 +15,40 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.DICOM_STORES;
+import static com.quantumsoft.qupathcloud.dao.Constants.LOCATIONS;
+import static com.quantumsoft.qupathcloud.dao.Constants.PROJECTS;
+import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * DicomStoresPathBuilder to build a path for access to DICOM Stores in chosen Project, Location and
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/DicomStoresPathBuilder.java]: DicomStoresPathBuilder builds a path to access DICOM Stores in the specified Project, Location and

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/FramePathBuilder.java [hunk 6]
```diff
@@ -15,43 +15,62 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.SERIES;
+import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
+import static com.quantumsoft.qupathcloud.dao.Constants.STUDIES;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * PFramePathBuilder to build a path for access to a frame in chosen Project, Location, Dataset,
+ * DICOM Store, Study, Series, and Instance.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/FramePathBuilder.java:33]: PFramePathBuilder builds a path to access a frame in the specified Project, Location, Dataset, DICOM Store, Study, Series, and Instance.

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/InstancesPathBuilder.java [hunk 7]
```diff
@@ -15,38 +15,58 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.STUDIES;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * InstancesPathBuilder to build a path for access to Instances in chosen Project, Location,
+ * Dataset, DICOM Store, Study and Series. If Study and Series are null then will be built a path
+ * for all Instances in a chosen Project, Location, Dataset, and DICOM Store.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/InstancesPathBuilder.java:34]: InstancesPathBuilder builds a path to access Instances in the specified Project, Location,
 Dataset, DICOM Store, Study and Series. If the Study and Series are null then the path will be built
 for all Instances in a chosen Project, Location, Dataset, and DICOM Store.

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/LocationsPathBuilder.java [hunk 8]
```diff
@@ -19,14 +19,23 @@
 import static com.quantumsoft.qupathcloud.dao.Constants.PROJECTS;
 import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
 
-public class LocationsPathBuilder implements PathBuilder{
-    private QueryBuilder queryBuilder;
+/**
+ * LocationsPathBuilder to build a path to access all locations in the selected project.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/LocationsPathBuilder.java]: LocationsPathBuilder builds a path to access all locations in the selected project.

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/SeriesPathBuilder.java [hunk 9]
```diff
@@ -15,37 +15,55 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.STUDIES;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * SeriesPathBuilder to build a path for access to Series in chosen Project, Location, Dataset,
+ * DICOM Store and Study. If Study is null then will be built a path for all Series in chosen
+ * Project, Location, Dataset, and DICOM Store.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/SeriesPathBuilder.java:33]: SeriesPathBuilder builds a path to access the Series in the specified Project, Location, Dataset, DICOM Store and Study. If Study is null, then the path will be built for all Series in the specified Project, Location, Dataset, and DICOM Store.

File: src/main/java/com/quantumsoft/qupathcloud/dao/spec/StudiesPathBuilder.java [hunk 10]
```diff
@@ -15,29 +15,46 @@
...
+import static com.quantumsoft.qupathcloud.dao.Constants.STAGE;
+import static com.quantumsoft.qupathcloud.dao.Constants.STUDIES;
+
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
 
-import static com.quantumsoft.qupathcloud.dao.Constants.*;
+/**
+ * StudiesPathBuilder to build a path for access to Studies in chosen Project, Location, Dataset and
+ * DICOM Store.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/dao/spec/StudiesPathBuilder.java:31]: StudiesPathBuilder builds a path to access Studies in a specified Project, Location, Dataset and DICOM Store.

File: src/main/java/com/quantumsoft/qupathcloud/entities/Dataset.java [hunk 11]
```diff
@@ -18,62 +18,108 @@
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import java.util.Objects;
 
+/**
+ * The type Dataset.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/entities/Dataset.java]: This does not really add any additional information - specify that the Dataset represents or remove the comment...

File: src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfFrameContentSequence.java [hunk 12]
```diff
@@ -18,15 +18,29 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.quantumsoft.qupathcloud.entities.DicomAttribute;
 
+/**
+ * The type Object of Frame Content Sequence.
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfFrameContentSequence.java]: Same here as above - specify what this class represents..

File: src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfPlanePositionSequence.java [hunk 13]
```diff
@@ -18,25 +18,51 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.quantumsoft.qupathcloud.entities.DicomAttribute;
 
+/**
+ * The type Object of Plane Position Sequence.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfPlanePositionSequence.java:26]: Same here..

File: src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfPerframeFunctionalGroupsSequence.java [hunk 14]
```diff
@@ -18,25 +18,51 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.quantumsoft.qupathcloud.entities.DicomAttribute;
 
+/**
+ * The type Object of Per-frame Functional Groups Sequence.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/entities/instance/objects/ObjectOfPerframeFunctionalGroupsSequence.java:26]: Same here..

File: src/main/java/com/quantumsoft/qupathcloud/entities/instance/Instance.java [hunk 15]
```diff
@@ -21,175 +21,322 @@
...
 import com.quantumsoft.qupathcloud.entities.instance.objects.ObjectOfPerframeFunctionalGroupsSequence;
 import com.quantumsoft.qupathcloud.exception.QuPathCloudException;
-
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+/**
+ * The type Instance.
+ */
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/entities/instance/Instance.java:30]: Same here..

**Review by poojavenkatram — COMMENTED** [review]
Thanks Artyom this is must easier to review :) Just a few small comments..

**Comment by arvgord** [comment]
Thanks for the review! I've changed all the points you mentioned.

**Review by poojavenkatram — APPROVED** [review]
Thanks for the changes Artyom :)

**Comment by poojavenkatram** [comment]
Can you also add the checker which will automatically check if someone is trying to check in the code which are not formatted with the Google Style Guide?

**Comment by arvgord** [comment]
Unfortunately, I haven't found an automatic check for Java only for [C++](https://google.github.io/styleguide/).

**Comment by arvgord** [comment]
Do I need to add changes for this PR?

**Comment by poojavenkatram** [comment]
We can add it in a follow up - my concern was that if I merge this and a follow up PR doesn't format with the Style guide the purpose of doing this is kind of lost..I am taking a look into what tools are available for the checking as well - will keep you posted. I will merge this and once we have a tool we can have a follow up PR to add the checker. Thanks :)

**Comment by arvgord** [comment]
I will create another two PRs with code style changes. Because not all places in the current version use Google Code Style. #13

**Pull request merged**